server/app.js

```diff
@@ -37,4 +39,176 @@ io.on('connection', (socket) => {
+        joinTable(io, socket, msg);
+    });
+
+    // each player will send a message upon a game play button click, this will check whether or not it is the correct player's turn and if the move is valid
+    socket.on('turnDecision', function(msg){
+        turnDecision(io, socket, msg);
+    });
+
+});
+
+
+// a list of sample profiles already put into the system, these are not the real profiles (just hard coded ones)
+// need to update to use the real profiles stored in the server
+let profiles = [
+    {userID: 1000, name: 'Mark123', chips: 20000, icon: 'player_icon_1'},
+    {userID: 2000, name: 'John456', chips: 40000, icon: 'player_icon_1'},
+    {userID: 3000, name: 'Luke854', chips: 60000, icon: 'player_icon_1'},
+    {userID: 4000, name: 'Jane027', chips: 80000, icon: 'player_icon_1'},
+    {userID: 5000, name: 'Stan693', chips: 10000, icon: 'player_icon_1'},
+    {userID: 6000, name: 'Mack273', chips: 20, icon: 'player_icon_1'},
+];
+// the entire list of tables, might need a better way to store them?
+let tables = [];
+tables.push(new PokerTable()); // initially we will have one table, when socket rooms are implemented then multiple tables will be added
+
+
+// these next three functions are for signing up a client to the poker table, need to use the real login and tables page to do this
+function login(socket, msg){
+    let profile = profiles.find(profile => profile.userID === msg);
+    if(profile === undefined){// the user does not exist
+        return;
+    }
+    else{// send the current table state
+        console.log('emitted');
+        socket.emit('tableState', JSON.stringify(tables[0].getTableState()));
+    }
+}
+function canUserJoinTable(tableToJoin, profile){
+    if(tableToJoin === undefined || tableToJoin.isTableFull() || tableToJoin.isPlayerAtTable(profile) || profile.chips < tableToJoin.buyIn){
+        return false; // the user cannot join this table
+    }
+    else{
+        return true;
+    }
+}
+function joinTable(io, socket, msg){
+    let profile = profiles.find(profile => profile.userID === msg.userID); // get their profile
+    let tableToJoin = tables.find(table => table.tableID === msg.tableID); // get the table they wish to join
+    if(canUserJoinTable(tableToJoin, profile)){// the user can join the table
+        let seatID = tableToJoin.addPlayerToTable(profile, socket.id); // the table will fill a seat based on the profile and send back the seat id
+        socket.emit('yourSeatID', seatID); // send him back his tableSeatID
+        io.emit('tableState', JSON.stringify(tableToJoin.getTableState())); // send to everyone the updated table state
+        if(tableToJoin.canAGameBegin() && !tableToJoin.tableActive){ // check if we can begin a game
+            tableToJoin.tableActive = true; // the table is now active
+            beginTheGame(io, tableToJoin);
+        }
+    }
+    else{
+        return; // the user can not join the table
```

Comment:
Should we return some sort of error socket.io response, telling them that they were unable to join?

Reply:
i guess we haven't discussed that yet, but from my understanding that would be a message sent to the tables page (that whole function would be used to talk to the tables page)

Reply:
I agree, I should have this covered in the tables page so they don't reach the poker table at all if they cannot join.

Reply:
If that's the case, is `else` even needed? The function will just return normally even without `else`

Reply:
removed